### PR TITLE
Added description to Exercise03AI

### PR DESCRIPTION
**Pull request by piuswalter**

### CODE REVIEW

File: src/Exercise03AI.ipynb [hunk 1]
```diff
@@ -37,10 +37,15 @@
...
-    "Alle restlichen Funktionen werden von der Basisklasse `ChessAI` vererbt. "
+    "Diese Klasse implementiert die Simplified Evaluation Function zur Berechnung des nächsten Zuges im Mittelspiel. Hierbei wird die `minimax`-Funktion der Super-Klasse (`Exercise02AI`) aufgerufen und durch die nun implementierte Simplified Evaluation Function verbessert.\n",
+    "\n",
+    "Hierzu werden zu Beginn die in der [Simplified Evaluation Function](https://www.chessprogramming.org/Simplified_Evaluation_Function#Piece-Square_Tables) beschriebenen `Piece-Square Tables` definiert. Diese weisen jeder der 64 möglichen Positionen auf dem Schachfeld für alle der Schachfiguren (Bauer, Springer, Läufer, Turm, Dame und König) eine numerische Bewertung in Form einer Ganzzahl zu. Diese kann sowohl positiv (gute Position für die Figur) oder negativ (schlechte Position für die Figur) sein und wird als Liste dargestellt. Theoretisch verfügt der König über zwei verschiedene Listen, eine für das Mittelspiel, eine für das Endspiel. Da das Endspiel aber von der Syzygy-Bibliothek übernommen wird, fällt diese weg und es reicht die Implementierung der List für das Mittelspiel.\n",
```
Comment (p-fruck) [inline src/Exercise03AI.ipynb:44]:
```suggestion
    "Hierzu werden zu Beginn die in der [Simplified Evaluation Function](https://www.chessprogramming.org/Simplified_Evaluation_Function#Piece-Square_Tables) beschriebenen `Piece-Square Tables` definiert. Diese weisen jeder der 64 möglichen Positionen auf dem Schachfeld für alle der Schachfiguren (Bauer, Springer, Läufer, Turm, Dame und König) eine numerische Bewertung in Form einer Ganzzahl zu. Diese kann sowohl positiv (gute Position für die Figur) oder negativ (schlechte Position für die Figur) sein und wird als Liste dargestellt. Theoretisch verfügt der König über zwei verschiedene Listen, eine für das Mittelspiel, eine für das Endspiel. Da das Endspiel aber von der Syzygy-Bibliothek übernommen wird, fällt diese weg und es reicht die Implementierung der Liste für das Mittelspiel.\n",
```

File: src/Exercise03AI.ipynb [hunk 2]
```diff
@@ -37,10 +37,15 @@
...
+    "\n",
+    "Anhand der Piece-Square Table der Bauern `PST_PAWN` die im Folgenden definiert ist, wird ersichtlich, dass Bauern auf den Felder `f2`, `g2` und `h2` (symmetrischer Weise natürlich auch auf `a2`, `b2` und `c2`) Boni (positive Bewertung) erhalten, während sie auf den Felder `f3` und `g3` Abzüge (negative Bewertungen) erhalten.\n",
```
Comment (p-fruck) [inline src/Exercise03AI.ipynb:46]:
```suggestion
    "Anhand der Piece-Square Table der Bauern `PST_PAWN`, die im Folgenden definiert ist, wird ersichtlich, dass Bauern auf den Feldern `f2`, `g2` und `h2` (symmetrischer Weise natürlich auch auf `a2`, `b2` und `c2`) Boni (positive Bewertung) erhalten, während sie auf den Felder `f3` und `g3` Abzüge (negative Bewertungen) erhalten.\n",
```

**Review by p-fruck — CHANGES_REQUESTED** [review]

File: src/Exercise03AI.ipynb (hunk 1 above)
Comment (antonplagemann) [inline src/Exercise03AI.ipynb:44]:
```suggestion
    "Hierzu werden zu Beginn die in der [Simplified Evaluation Function](https://www.chessprogramming.org/Simplified_Evaluation_Function#Piece-Square_Tables) beschriebenen `Piece-Square Tables` definiert. Diese weisen jeder der 64 möglichen Positionen auf dem Schachfeld für alle der Schachfiguren (Bauer, Springer, Läufer, Turm, Dame und König) eine numerische Bewertung in Form einer Ganzzahl zu. Diese kann sowohl positiv (gute Position für die Figur) oder negativ (schlechte Position für die Figur) sein und wird als Liste dargestellt. Theoretisch verfügt der König über zwei verschiedene Listen, eine für das Mittelspiel, eine für das Endspiel. Da das Endspiel aber von der Syzygy-Bibliothek übernommen wird, fällt die Endpiel-Liste weg und es reicht die Implementierung der Liste für das Mittelspiel.\n",
```

File: src/Exercise03AI.ipynb (hunk 2 above)
Comment (antonplagemann) [inline src/Exercise03AI.ipynb:46]:
```suggestion
    "Anhand der Piece-Square Table der Bauern `PST_PAWN` die im Folgenden definiert ist, wird beispielsweise ersichtlich, dass Bauern auf den Felder `f2`, `g2` und `h2` (symmetrischer Weise natürlich auch auf `a2`, `b2` und `c2`) Boni (positive Bewertung) erhalten, während sie auf den Felder `f3` und `g3` Abzüge (negative Bewertungen) erhalten.\n",
```

File: src/Exercise03AI.ipynb [hunk 3]
```diff
@@ -155,6 +167,28 @@
     "            return sum([-(self.PIECE_TO_PST[piece][chess.square_mirror(i)]) for i in board.pieces(piece, color)])"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Die Funktion `evaluate` berechnet den Wert eines gesamten Boards, welches als Argument übergeben wird. Als Grundlage werden die Materialwerte der einzelnen Figuren in Centipawns benötigt und wie folgt gesetzt um in der Elternklasse überschrieben zu werden:\n",
```
Comment (antonplagemann) [inline src/Exercise03AI.ipynb:174]:
```suggestion
    "Die Funktion `evaluate` berechnet den Wert eines gesamten Boards, welches als Argument übergeben wird. Als Grundlage werden die Materialwerte der einzelnen Figuren in Centipawns benötigt, weshalb die bereits in der Elternklasse (`Exercise02AI`) gesetzten Werte wie folgt überschrieben werden:\n",
```

File: src/Exercise03AI.ipynb [hunk 4]
```diff
@@ -155,6 +167,28 @@
...
+    "|---|---|\n",
+    "| Bauer (pawn) | 100  |\n",
+    "| Springer (knight) | 320  |\n",
+    "| Läufer (bishop) | 330  |\n",
+    "| Turm (rook) | 500  |\n",
+    "| Dame (queen) | 900  |\n",
+    "\n",
+    "Die Funktion `evaluate` berechnet den Wert des Boards wie folgt:\n",
+    "\n",
+    "1. Berechnung aller weißen und schwarzen Figuren mithilfe der Funktion `calculate_pst_value`\n",
```
Comment (antonplagemann) [inline src/Exercise03AI.ipynb:186]:
```suggestion
    "1. Berechnung der Anzahl aller weißen und schwarzen Figuren mithilfe der Funktion `calculate_pst_value`\n",
```

**Review by antonplagemann — CHANGES_REQUESTED** [review]
Nice 🙌

File: src/Exercise03AI.ipynb [hunk 5]
```diff
@@ -41,9 +41,9 @@
...
+    "Hierzu werden zu Beginn die in der [Simplified Evaluation Function](https://www.chessprogramming.org/Simplified_Evaluation_Function#Piece-Square_Tables) beschriebenen `Piece-Square Tables` definiert. Diese weisen jeder der 64 möglichen Positionen auf dem Schachfeld für alle der Schachfiguren (Bauer, Springer, Läufer, Turm, Dame und König) eine numerische Bewertung in Form einer Ganzzahl zu. Diese kann sowohl positiv (gute Position für die Figur) oder negativ (schlechte Position für die Figur) sein und wird als Liste dargestellt. Theoretisch verfügt der König über zwei verschiedene Listen, eine für das Mittelspiel, eine für das Endspiel. Da das Endspiel aber von der Syzygy-Bibliothek übernommen wird, fällt diese weg und es reicht die Implementierung der Liste für das Mittelspiel.\n",
```
Comment (antonplagemann) [inline src/Exercise03AI.ipynb:44]:
```suggestion
    "Hierzu werden zu Beginn die in der [Simplified Evaluation Function](https://www.chessprogramming.org/Simplified_Evaluation_Function#Piece-Square_Tables) beschriebenen `Piece-Square Tables` definiert. Diese weisen jeder der 64 möglichen Positionen auf dem Schachfeld für alle der Schachfiguren (Bauer, Springer, Läufer, Turm, Dame und König) eine numerische Bewertung in Form einer Ganzzahl zu. Diese kann sowohl positiv (gute Position für die Figur) oder negativ (schlechte Position für die Figur) sein und wird als Liste dargestellt. Theoretisch verfügt der König über zwei verschiedene Listen, eine für das Mittelspiel, eine für das Endspiel. Da das Endspiel aber von der Syzygy-Bibliothek übernommen wird, fällt die Endpiel-Liste weg und es reicht die Implementierung der Liste für das Mittelspiel.\n",
```

**Review by antonplagemann — CHANGES_REQUESTED** [review]

**Review by antonplagemann — APPROVED** [review]

**Review by p-fruck — APPROVED** [review]

**Pull request merged**